tests/admin_checks/tests.py

```diff
@@ -157,18 +167,24 @@ def test_middleware_dependencies(self):
         errors = admin.checks.check_dependencies()
         expected = [
             checks.Error(
-                "'django.contrib.auth.middleware.AuthenticationMiddleware' "
+                "'django.contrib.auth.middleware.AuthenticationMiddleware' (or a subclass) "
                 "must be in MIDDLEWARE in order to use the admin application.",
                 id='admin.E408',
             ),
             checks.Error(
-                "'django.contrib.messages.middleware.MessageMiddleware' "
+                "'django.contrib.messages.middleware.MessageMiddleware' (or a subclass) "
                 "must be in MIDDLEWARE in order to use the admin application.",
                 id='admin.E409',
             )
         ]
         self.assertEqual(errors, expected)
 
+        # Subclasses are permitted
```

Comment:
This doesn't depend on anything above, so use a separate test: `test_middleware_subclasses`

Reply:
Hehe, I initially did, however @carltongibson made the comment that perhaps they should be merged – https://github.com/django/django/pull/11057#discussion_r264271825

Reply:
Sorry. 😬 I thought they made more sense together, but let's go with what Tim said. (Would have needed a full-stop anyhow...)